src/dotnet/Audio.Service/AudioProcessor.cs

```diff
@@ -84,66 +84,66 @@ public AudioProcessor(Options settings, IServiceProvider services)
                 TimeSpan.Zero,
                 AudioSourceLog,
                 cancellationToken);
-            var openSegment = new OpenAudioSegment(0, record, audio, author, languages, OpenAudioSegmentLog);
+            var openSegment = new OpenAudioSegment(0,
+                record,
+                audio,
+                author,
+                languages,
+                OpenAudioSegmentLog);
+            var recordedAt = Moment.EpochStart + TimeSpan.FromSeconds(record.ClientStartOffset);
+            openSegment.SetRecordedAt(recordedAt);
             streamId = openSegment.StreamId;
-            openSegment.SetRecordedAt(Moment.EpochStart + TimeSpan.FromSeconds(record.ClientStartOffset));
 
             var audioStream = openSegment.Audio
                 .GetFrames(cancellationToken)
                 .Select(f => f.Data);
-
             var publishAudioTask = BackgroundTask.Run(
                 () => AudioStreamServer.Write(openSegment.StreamId, audioStream, cancellationToken),
                 Log,
                 $"{nameof(AudioStreamServer.Write)} failed",
                 cancellationToken);
 
             var audioEntryTask = BackgroundTask.Run(
-                () => CreateAudioEntry(openSegment, cancellationToken),
+                () => CreateAudioEntry(openSegment, recordedAt, cancellationToken),
                 Log,
                 $"{nameof(CreateAudioEntry)} failed",
                 cancellationToken);
 
             var transcribeTask = BackgroundTask.Run(
-                () => TranscribeAudio(record.ChatId, openSegment, audioEntryTask, cancellationToken),
+                () => TranscribeAudio(record.ChatId, openSegment, audioEntryTask, CancellationToken.None),
```

Comment:
IMO makes sense to still pass CT here, but process all of them gracefully by shutting down source audio stream once the originally passed token gets cancelled.

Reply:
Maybe we should actually using "wrapped" `audioStream` - i.e. call something after `GetFrames` that will wrap it into a normally terminating stream on cancellation.